test/syncenginetestutils.h

```diff
@@ -706,17 +706,26 @@ class FakeErrorReply : public QNetworkReply
 
 class FakeQNAM : public QNetworkAccessManager
 {
+public:
+    using Observer = std::function<void(QByteArray verb, QString fileName)>;
```

Comment:
Would it be not more generic to have the whole QNetworkRequest as a parameter? 
So we could also test "Does the call contains the header FooBar"